haskell_scotty/ClientRepo.hs

```diff
@@ -16,3 +18,19 @@ getClients = do
 
 instance QueryResults Client where
   convertResults fs vs = Client $... zip fs vs
+
+saveClient :: Client -> IO Client
+saveClient client = do
+  conn <- connectDB
+  execute conn "INSERT INTO clients \
+    \(name,address,city,state,zip,country,email,contact,phone,created_at,updated_at) values \
+    \(?,?,?,?,?,?,?,?,?,?,?)" client
+  results <- query_ conn "SELECT LAST_INSERT_ID()"
+  let [Only clientId] = results
+  return client { Client.id = clientId }
+
+instance QueryParams Client where
+  renderParams
+    (Client _ name address city state zip country email contact phone created_at updated_at) =
+      [render name, render address, render city, render state, render zip, render country, render email, render contact, render phone, render created_at, render updated_at]
```

Comment:
Yeah, if you try that the last couple of elements of the list are of different type than the first ones, which is not allowed by the type system. You could wrap them in a constructor and do stuff, but that's not any better than what you have. You could also replicate the magic I built earlier to get this to work. All of it is ugly, for sure.